comp/core/workloadmeta/collectors/internal/ecsfargate/ecsfargate.go

```diff
@@ -87,9 +88,14 @@ func (c *collector) setTaskCollectionParser() {
 	var err error
 	c.metaV4, err = ecsmeta.V4FromCurrentTask()
 	if c.taskCollectionEnabled && err == nil {
+		log.Infof("detailed task collection enabled, using metadata v4 endpoint")
 		c.taskCollectionParser = c.parseTaskFromV4Endpoint
 		return
+	} else if c.taskCollectionEnabled && err != nil {
+		log.Warnf("failed to initialize metadata v4 client, using metdata v2: %v", err)
 	}
```

Comment:
similar to previous comment.

Having something like the following makes the code easier to read and understand:

```
if c.taskCollectionEnabled {
    var err error
	c.metaV4, err = ecsmeta.V4FromCurrentTask()
	if err == nil {
		log.Infof("detailed task collection enabled, using metadata v4 endpoint")
		c.taskCollectionParser = c.parseTaskFromV4Endpoint
		return
	} 
	log.Warnf("failed to initialize metadata v4 client, using metdata v2: %v", err)
}
```

it also avoids calling `ecsmeta.V4FromCurrentTask()` when task collection is not enabled.